rllib/models/catalog.py

```diff
@@ -174,6 +174,16 @@
     # your environment instead to preprocess observations.
     "custom_preprocessor": None,
 
+    # === Options for ModelConfigs in RLModules ===
+    # The latent dimension to encode into.
+    # Since most RLModules have an encoder and heads, this establishes an agreement
+    # on the dimensionality of the latent space they share.
+    # This has no effect for models outside RLModule.
+    # If None, model_config['fcnet_hiddens'][-1] value will be used to guarantee
+    # backward compatibility to old configs. This yields different models than past
+    # versions of RLlib.
+    "encoder_latent_dim": None,
+
```

Comment:
@kouroshHakha We can easily do MLP Encoders and simply set the output_dim to the last fcnet_hiddens dim to yield the same models as we did prior to our Encoder/Head architecture.

For what we describe today with a CNN and Heads, we had the VisionNet class before. That class didn't need to "know" the latent dim at configuration time, because it would do an "experimental" forward pass by doing observation_space->CNN->flatten->latent_space and constructing the _post_fc_net_hidden_ layers from there.
Today, we can't do that because we want to know the latent dim at configuration time for spec checking and overall cleaner separation between heads and encoder. That's why, afaics, we need this new parameter.